src/pages/workspace/workflows/ToggleSettingsOptionRow.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {useState} from 'react';
+import {View} from 'react-native';
+import type {SvgProps} from 'react-native-svg';
+import Icon from '@components/Icon';
+import Switch from '@components/Switch';
+import Text from '@components/Text';
+import useNetwork from '@hooks/useNetwork';
+import useThemeStyles from '@hooks/useThemeStyles';
+
+type ToggleSettingOptionRowProps = {
+    icon: React.FC<SvgProps>;
+    title: string;
+    subtitle: string;
+    hasBeenToggled: boolean;
+    onToggle: (isEnabled: boolean) => void;
+    subMenuItems?: React.ReactNode;
+};
+
+function ToggleSettingOptionRow({icon, title, subtitle, onToggle, subMenuItems, hasBeenToggled}: ToggleSettingOptionRowProps) {
+    const [isEnabled, setIsEnabled] = useState(hasBeenToggled);
+    const [shouldShowOfflineStyle, setShouldShowOfflineStyle] = useState(false);
+    const styles = useThemeStyles();
+    const {isOffline} = useNetwork();
+    const ICON_SIZE = 48;
+    const toggleSwitch = () => {
+        if (isEnabled && isOffline) {
+            setShouldShowOfflineStyle(true);
+        } else {
+            setShouldShowOfflineStyle(false);
+        }
```

Comment:
It feels weird to keep state for this. Can we just apply the styles directly? Since we're connecting to the `useNetwork` hook, the component would re-render if there's a network status change, no?

Reply:
Actually we need to apply offline status to the `Switch` component whenever the user _toggles_  off the switch.

If we'd rely only on `isOffline`, it would apply the style automatically which is not what we want. We want the user action and the offline state combo to add the correct styling.

See https://docs.google.com/document/d/1Qv6V4nRqIbZfzQML_saCbrAEqQVIBsiQ3HE1UO7RCxQ/edit#bookmark=id.2jry5dd0c0a

Section (`Similarly, optimistically disabling Delayed Submission, Approvals, or Payments on the workflows page uses Pattern B`)

Reply:
Thank you, I'll keep it in mind for the future to use it for everything related to pattern B